x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -99,7 +99,76 @@ export type IndexPatternDimensionPanelProps = DatasourceDimensionPanelProps & {
 };
 
 export function IndexPatternDimensionPanel(props: IndexPatternDimensionPanelProps) {
-  return <div>Dimension Panel</div>;
+  const fields = props.state.indexPatterns[props.state.currentIndexPatternId].fields;
+  const columns: IndexPatternColumn[] = fields.map((field, index) => ({
+    operationId: `${index}`,
+    label: `Value of ${field.name}`,
+    dataType: field.type as DataType,
+    isBucketed: false,
+
+    operationType: 'value',
+  }));
+
+  const filteredColumns = columns.filter(col => {
+    const { operationId, label, dataType, isBucketed } = col;
+
+    return props.filterOperations({
+      id: operationId,
+      label,
+      dataType,
+      isBucketed,
+    });
+  });
+
+  let selectedColumn: IndexPatternColumn | null = null;
```

Comment:
This seems like it'd be better as a const + ternary operation. I find that let statements require more mental stack space than consts. I have to track the rest of the function to determine if it changes anywhere outside of its initialization. In this case, it's essentially a const (only set at initialization). So I'd change it to a const.

Reply:
I can change this to a const, but it's simpler as a short-circuited boolean operation.